6_survival_analysis/nbconverted/survival_hit_groups.py

```diff
@@ -0,0 +1,704 @@
+
+def show_values_on_bars(ax):
+    for i in range(plot_df.shape[0]):
+        _x = i
+        _y = plot_df.loc[i, '{}_diff'.format(metric)]
+        val = plot_df.loc[i, 'identifier']
+        if _y > 0:
+            ax.text(_x, _y + 10, val, ha="center", rotation=90) 
+        else:
+            ax.text(_x, _y - 100, val, ha="center", rotation=90)
+            
+axarr[1].set_ylim(-220, 120)
+show_values_on_bars(axarr[1])
+
+plt.tight_layout()
+
+
+# In[25]:
+
+
+sns.set({'figure.figsize': (12, 6)})
+sns.set_style('whitegrid')
+
+fig, axarr = plt.subplots(1, 2)
+
+identifier = 'RB1_LIHC'
+cancer_classification = 'TSG'
+hits_classification = 'two'
+
+id_clinical_df = get_groups_for_gene_and_tissue(
+    identifier, cancer_classification, hits_classification)
+
+print(identifier, r'delta TS:', '{:.4f}'.format(class_2_surv_df.loc[class_2_surv_df.identifier == identifier, 'ts_diff'].to_numpy()[0]))
+plot_id(identifier, id_clinical_df)
+
+
+# So we can see that the "test statistic distribution" is centered around 0, meaning that in most cases either both classes distinguish similarly between survival groups, or neither do. If the "class 2" made sense as a survival marker we'd expect to see the distribution shifted to the right a bit, and we don't really see that.
+# 
+# In some cases we do see a bit of a survival difference, but most of them aren't huge, and many are likely affected by sample size (in other words, if we only have a few two-hit samples the test statistic is never going to be large, e.g. PTEN_LGG).
+
+# ### Run for all "class 1" tumor suppressors
+# 
+# In Park et al, "class 1" tumor suppressors are "one-hit" genes, meaning a single mutation (point mutation or CNV) should contribute to cancer initiation/development.
+# 
+# Like before, we'll look at the difference between 1+ hits vs 0, and 2 hits vs. 0 or 1. Positive test statistic differences indicates that the one-hit classification gives "better" survival groups, and vice-versa.
+
+# In[26]:
+
+
+class_1_ids = (
+    park_loss_df[park_loss_df.classification == 'TSG']
+      .index.unique()
+      .difference(park_loss_sig_df.index)
+)
+    
+print(len(class_1_ids))
+print(class_1_ids[:10])
+
+
+# In[27]:
+
+
+class_1_surv_df = []
+columns = [
+    'identifier',
+    'n_mutant',
+    'n_wildtype',
+    'chisq',
+    'p_val',
+    'n_mutant_alt',
+    'n_wildtype_alt',
+    'chisq_alt',
+    'p_val_alt',
+]
+
+for identifier in class_1_ids:
+    
+    results = [identifier]
+    
+    id_clinical_df = get_groups_for_gene_and_tissue(
+        identifier, 'TSG', 'one')
+    
+    for ix, mut_col in enumerate(['is_mutated', 'is_mutated_alt']):
+    
+        n_mutant = id_clinical_df[mut_col].sum()
+        n_wildtype = id_clinical_df.shape[0] - n_mutant
+        
+        if n_mutant < 2 or n_wildtype < 2:
+            # statistical testing won't work in this case so just continue
+            print(identifier, mut_col, n_mutant, n_wildtype, file=sys.stderr)
```

Comment:
Is there a reason you're printing it out this way instead of just having it in a dataframe and printing from that?

Reply:
Not a good reason - I mainly just wanted to get an idea of how often this is happening. A dataframe would definitely accomplish that, but I think this works fine too and is slightly less work.